src/js/components/DateInput/stories/Range.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { Grommet, Box, DateInput } from 'grommet';
+import { grommet } from 'grommet/themes';
+
+const now = new Date();
+const lastWeek = new Date(now);
+lastWeek.setDate(lastWeek.getDate() - 7);
+
+const dateFormat = new Intl.DateTimeFormat(undefined, {
+  month: 'short',
+  day: 'numeric',
+});
+
+const Example = () => {
+  const [value, setValue] = React.useState([
+    lastWeek.toISOString(),
+    now.toISOString(),
+  ]);
+  const onChange = event => {
+    const nextValue = event.value;
+    console.log('onChange', nextValue);
+    setValue(nextValue);
+  };
+  return (
+    <Grommet theme={grommet}>
+      <Box align="center" pad="large">
+        <DateInput
+          value={value}
+          buttonProps={{
+            label: `${dateFormat.format(
+              new Date(value[0]),
+            )} - ${dateFormat.format(new Date(value[1]))}`,
+          }}
```

Comment:
A nice enhancement request will be to make it this label built-in into the component.

Reply:
Maybe later. That opens up some more layout flavors and this component already feels like it has a lot.